pkg/aws/awscontrolplane/validate_awscontrolplane.go

```diff
@@ -53,39 +53,45 @@ func NewValidator(config config.Config) (*Validator, error) {
 
 func (v *Validator) Validate(request *admissionv1.AdmissionRequest) (bool, error) {
 	var awsControlPlane infrastructurev1alpha2.AWSControlPlane
-	var azReplicaMatches bool
 	var err error
 
 	if _, _, err := validator.Deserializer.Decode(request.Object.Raw, nil, &awsControlPlane); err != nil {
 		return false, microerror.Maskf(parsingFailedError, "unable to parse awscontrol plane: %v", err)
 	}
-	azAllowed, err := v.AZValid(awsControlPlane)
+	_, err = v.AZValid(awsControlPlane)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
-	azCountAllowed, err := v.AZCount(awsControlPlane)
+	_, err = v.AZCount(awsControlPlane)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
-	instanceTypeAllowed, err := v.InstanceTypeValid(awsControlPlane)
+	_, err = v.InstanceTypeValid(awsControlPlane)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
-	controlPlaneLabelMatches, err := v.ControlPlaneLabelMatch(awsControlPlane)
+	_, err = v.AZOrder(request)
+	if err != nil {
+		return false, microerror.Mask(err)
+
+	}
+	_, err = v.AZUnique(awsControlPlane)
+	if err != nil {
+		return false, microerror.Mask(err)
+
+	}
+	_, err = v.ControlPlaneLabelMatch(awsControlPlane)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
 	// when updating from single to HA validation of AZ replicas has to be ignored
 	if request.Operation == aws.CreateOperation {
-		azReplicaMatches, err = v.AZReplicaMatch(awsControlPlane)
+		_, err = v.AZReplicaMatch(awsControlPlane)
 		if err != nil {
 			return false, microerror.Mask(err)
 		}
-	} else {
-		azReplicaMatches = true
 	}
-
-	return controlPlaneLabelMatches && azAllowed && azCountAllowed && azReplicaMatches && instanceTypeAllowed, nil
+	return true, nil
```

Comment:
I do not follow how this works now. There is no case where there is `false, nil` returned. I consider this a behaviour change and it is weird that "everything is still fine". The only cases where `false` is returned now is when an error is returned. At least this is my understanding right now. Can you confirm?

Reply:
In case we reject the request in any of the validators we always throw an error. So in case any of the checks return `false` we won`t reach this last return.

Reply:
In this case the interface of the checker methods is confusing. They should then only return an error because the boolean is kind of useless as it seems. What do you think?

Reply:
Yeah I agree. We could let them only return an error so if they are nil it means everything is fine. Like this we can not run into issues with a potential `false, nil` case. It would also reinforce that there has to be a reason given if the validation fails.